lib/application.js

```diff
@@ -158,6 +158,11 @@ module.exports = class Application extends Emitter {
       keys: this.keys,
       secure: request.secure
     });
+    if (req && req.headers) {
+      request.ip = request.ips[0] || req.socket.remoteAddress || '';
+    } else {
+      request.ip = ''; // We are in a badly written test :P
```

Comment:
This could remove rate limiting if the attacker doesn't need the result of the request. I'd prefer we don't handle the request. However, the impact is minimal so I'll change it back to `|| ''` unless anybody else has a comment.